libsolidity/codegen/ExpressionCompiler.cpp

```diff
@@ -1252,13 +1252,16 @@ bool ExpressionCompiler::visit(FunctionCall const& _functionCall)
 			auto const functionPtr = dynamic_cast<FunctionTypePointer>(arguments[0]->annotation().type);
 			if (function.kind() == FunctionType::Kind::ABIEncodeCall)
 			{
-				solAssert(arguments.size() == 2, "");
-				solAssert(functionPtr, "");
+				solAssert(arguments.size() == 2);
+				solAssert(functionPtr);
 
+				auto const tupleExpression = dynamic_pointer_cast<TupleExpression const>(arguments[1]);
 				auto const tupleType = dynamic_cast<TupleType const*>(arguments[1]->annotation().type);
 
+				solAssert(!tupleType || tupleExpression, "Tuple type without tuple expression?!");
+
 				// Account for tuples with one component which become that component
-				if (!tupleType)
+				if (!tupleType || tupleExpression->isInlineArray())
```

Comment:
TupleType is only for tuples. So if the tuple expression is an inline array. its type will not be a tuple type.